RUFAS/routines/field/crop/nitrogen_incorporation.py

```diff
@@ -396,7 +397,19 @@ def determine_nutrient_shape_parameters(
             If half_mature_heat_fraction equals mature_heat_fraction.
 
         """
+        info_map = {
+            "class": NitrogenIncorporation.__class__.__name__,
+            "function": NitrogenIncorporation.determine_nutrient_shape_parameters.__name__,
+        }
+        om = OutputManager()
         if mature_heat_fraction == half_mature_heat_fraction:  # leads to divide by 0
+            om.add_error(
```

Comment:
```suggestion
        if mature_heat_fraction == half_mature_heat_fraction:  # leads to divide by 0
            info_map = {
                "class": NitrogenIncorporation.__class__.__name__,
                "function": NitrogenIncorporation.determine_nutrient_shape_parameters.__name__,
            }
            om = OutputManager()
            om.add_error(
```
Since the OM is only used within this if block, it is very slightly more efficient to instantiate the OM in it.